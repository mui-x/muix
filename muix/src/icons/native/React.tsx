import createSvgIcon from './lib/create-svg-icon'

export default createSvgIcon(
  'react',
  'React',
  true
)