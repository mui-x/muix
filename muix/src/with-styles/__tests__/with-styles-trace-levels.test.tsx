/** @jsx platform.createElement */
import React from 'react'

import { platform } from 'reactxx-sheeter'

import { ts, traceComponentEx } from "reactxx-tests";

describe("WITH STYLES TRACE LEVELS", () => {
  it("trace 1 WEB", () => traceTest(true, 1));
  it("trace 1 NATIVE", () => traceTest(false, 1));
  it("trace 2 WEB", () => traceTest(true, 2));
  it("trace 2 NATIVE", () => traceTest(false, 2));
  it("trace 3 WEB", () => traceTest(true, 3));
  it("trace 3 NATIVE", () => traceTest(false, 3));
  it("trace 4 WEB", () => traceTest(true, 4));
  it("trace 4 NATIVE", () => traceTest(false, 4));
  it("trace 5 WEB", () => traceTest(true, 5));
  it("trace 5 NATIVE", () => traceTest(false, 5));
});

const traceTest = (isWeb: boolean, level: 1 | 2 | 3 | 4 | 5) => {
  window.__TRACELEVEL__ = level
  traceComponentEx(isWeb,
    {
      root: ts.view = {
        backgroundColor: 'gray'
      }
    },
    ({ classes, classNameX, toClassNames, styleX, children }) => {
      const root = toClassNames([classes.root, classNameX])
      return <div classNameX={root} styleX={styleX}>
        {children}
      </div>
    },
    Comp => <Comp classNameX={{ padding: 10 }} styleX={{ margin: 10 }}>
      Text
  </Comp>)
}
