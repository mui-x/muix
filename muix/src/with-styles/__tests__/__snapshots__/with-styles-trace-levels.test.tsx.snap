// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WITH STYLES TRACE LEVELS trace 1 NATIVE 1`] = `
<ReactAny
  style={
    Object {
      "backgroundColor": "gray",
      "margin": 10,
      "padding": 10,
    }
  }
>
  traceLevel=1
</ReactAny>
`;

exports[`WITH STYLES TRACE LEVELS trace 1 WEB 1`] = `
<ReactAny
  className="b a"
  style={
    Object {
      "margin": 10,
    }
  }
>
  traceLevel=1
</ReactAny>
`;

exports[`WITH STYLES TRACE LEVELS trace 2 NATIVE 1`] = `
<ReactAny
  style={
    Object {
      "backgroundColor": "gray",
      "margin": 10,
      "padding": 10,
    }
  }
  trace={
    Object {
      "backgroundColor": "gray /*TestComponent.sheet.root*/",
      "margin": "10 /*styleX*/",
      "padding": "10 /*classNameX*/",
    }
  }
>
  traceLevel=2
</ReactAny>
`;

exports[`WITH STYLES TRACE LEVELS trace 2 WEB 1`] = `
<ReactAny
  className="b a"
  style={
    Object {
      "margin": 10,
    }
  }
  trace={
    Array [
      ".b { padding:10px /*classNameX*/ }",
      ".a { background-color:gray /*TestComponent.sheet.root*/ }",
    ]
  }
>
  traceLevel=2
</ReactAny>
`;

exports[`WITH STYLES TRACE LEVELS trace 3 NATIVE 1`] = `
<TestComponent>
  <InnerStateComponent>
    <TestComponentCode>
      <ReactAny
        style={
          Object {
            "backgroundColor": "gray",
            "margin": 10,
            "padding": 10,
          }
        }
        trace={
          Object {
            "backgroundColor": "gray /*TestComponent.sheet.root*/",
            "margin": "10 /*styleX*/",
            "padding": "10 /*classNameX*/",
          }
        }
      >
        traceLevel=3
      </ReactAny>
    </TestComponentCode>
  </InnerStateComponent>
</TestComponent>
`;

exports[`WITH STYLES TRACE LEVELS trace 3 WEB 1`] = `
<TestComponent>
  <InnerStateComponent>
    <TestComponentCode>
      <ReactAny
        className="b a"
        style={
          Object {
            "margin": 10,
          }
        }
        trace={
          Array [
            ".b { padding:10px /*classNameX*/ }",
            ".a { background-color:gray /*TestComponent.sheet.root*/ }",
          ]
        }
      >
        traceLevel=3
      </ReactAny>
    </TestComponentCode>
  </InnerStateComponent>
</TestComponent>
`;

exports[`WITH STYLES TRACE LEVELS trace 4 NATIVE 1`] = `
<TestComponent
  classNameX={
    Object {
      "padding": 10,
    }
  }
  styleX={
    Object {
      "margin": 10,
    }
  }
>
  <InnerStateComponent>
    <TestComponentCode
      classNameX={
        Object {
          "list": Array [
            Object {
              "atomicArray": Array [
                "padding: 10 /*classNameX*/",
              ],
            },
          ],
        }
      }
      classes={
        Object {
          "root": Object {
            "list": Array [
              Object {
                "atomicArray": Array [
                  "backgroundColor: gray /*TestComponent.sheet.root*/",
                ],
              },
            ],
          },
        }
      }
      styleX={
        Object {
          "list": Array [
            Object {
              "atomicArray": Array [
                "margin: 10 /*styleX*/",
              ],
            },
          ],
        }
      }
    >
      <ReactAny
        style={
          Object {
            "backgroundColor": "gray",
            "margin": 10,
            "padding": 10,
          }
        }
        trace={
          Object {
            "backgroundColor": "gray /*TestComponent.sheet.root*/",
            "margin": "10 /*styleX*/",
            "padding": "10 /*classNameX*/",
          }
        }
      >
        traceLevel=4
      </ReactAny>
    </TestComponentCode>
  </InnerStateComponent>
</TestComponent>
`;

exports[`WITH STYLES TRACE LEVELS trace 4 WEB 1`] = `
<TestComponent
  classNameX={
    Object {
      "padding": "10px",
    }
  }
  styleX={
    Object {
      "margin": 10,
    }
  }
>
  <InnerStateComponent>
    <TestComponentCode
      classNameX={
        Object {
          "list": Array [
            Object {
              "atomicArray": Array [
                ".b { padding:10px /*classNameX*/ }",
              ],
            },
          ],
        }
      }
      classes={
        Object {
          "root": Object {
            "list": Array [
              Object {
                "atomicArray": Array [
                  ".a { background-color:gray /*TestComponent.sheet.root*/ }",
                ],
              },
            ],
          },
        }
      }
      styleX={
        Object {
          "margin": 10,
        }
      }
    >
      <ReactAny
        className="b a"
        style={
          Object {
            "margin": 10,
          }
        }
        trace={
          Array [
            ".b { padding:10px /*classNameX*/ }",
            ".a { background-color:gray /*TestComponent.sheet.root*/ }",
          ]
        }
      >
        traceLevel=4
      </ReactAny>
    </TestComponentCode>
  </InnerStateComponent>
</TestComponent>
`;

exports[`WITH STYLES TRACE LEVELS trace 5 NATIVE 1`] = `
<TestComponent
  classNameX={
    Object {
      "padding": 10,
    }
  }
  styleX={
    Object {
      "margin": 10,
    }
  }
>
  <InnerStateComponent
    pipeState={Object {}}
    pipelineState={
      Object {
        "componentId": 9,
        "displayName": "TestComponent",
        "pipeStates": Array [
          Object {
            "classes": Object {
              "root": Object {
                "list": Array [
                  Object {
                    "atomicArray": Array [
                      "backgroundColor: gray /*TestComponent.sheet.root*/",
                    ],
                  },
                ],
              },
            },
          },
          Object {
            "classNameX": Object {
              "list": Array [
                Object {
                  "atomicArray": Array [
                    "padding: 10 /*classNameX*/",
                  ],
                },
              ],
            },
            "classes": null,
            "codeProps": Array [
              Object {
                "children": "Text",
              },
              null,
            ],
            "styleX": Object {
              "list": Array [
                Object {
                  "atomicArray": Array [
                    "margin: 10 /*styleX*/",
                  ],
                },
              ],
            },
          },
          undefined,
          Object {},
        ],
        "props": Object {
          "children": "Text",
          "classNameX": Object {
            "padding": 10,
          },
          "styleX": Object {
            "margin": 10,
          },
        },
        "propsCode": Object {
          "children": "Text",
          "classNameX": Object {
            "list": Array [
              Object {
                "atomicArray": Array [
                  "padding: 10 /*classNameX*/",
                ],
              },
            ],
          },
          "classes": Object {
            "root": Object {
              "list": Array [
                Object {
                  "atomicArray": Array [
                    "backgroundColor: gray /*TestComponent.sheet.root*/",
                  ],
                },
              ],
            },
          },
          "styleX": Object {
            "list": Array [
              Object {
                "atomicArray": Array [
                  "margin: 10 /*styleX*/",
                ],
              },
            ],
          },
        },
        "uniqueId": "TestComponent *19",
      }
    }
  >
    <TestComponentCode
      classNameX={
        Object {
          "list": Array [
            Object {
              "atomicArray": Array [
                "padding: 10 /*classNameX*/",
              ],
            },
          ],
        }
      }
      classes={
        Object {
          "root": Object {
            "list": Array [
              Object {
                "atomicArray": Array [
                  "backgroundColor: gray /*TestComponent.sheet.root*/",
                ],
              },
            ],
          },
        }
      }
      styleX={
        Object {
          "list": Array [
            Object {
              "atomicArray": Array [
                "margin: 10 /*styleX*/",
              ],
            },
          ],
        }
      }
    >
      <ReactAny
        style={
          Object {
            "backgroundColor": "gray",
            "margin": 10,
            "padding": 10,
          }
        }
        trace={
          Object {
            "backgroundColor": "gray /*TestComponent.sheet.root*/",
            "margin": "10 /*styleX*/",
            "padding": "10 /*classNameX*/",
          }
        }
      >
        traceLevel=5
      </ReactAny>
    </TestComponentCode>
  </InnerStateComponent>
</TestComponent>
`;

exports[`WITH STYLES TRACE LEVELS trace 5 WEB 1`] = `
<TestComponent
  classNameX={
    Object {
      "padding": "10px",
    }
  }
  styleX={
    Object {
      "margin": 10,
    }
  }
>
  <InnerStateComponent
    pipeState={Object {}}
    pipelineState={
      Object {
        "componentId": 8,
        "displayName": "TestComponent",
        "pipeStates": Array [
          Object {
            "classes": Object {
              "root": Object {
                "list": Array [
                  Object {
                    "atomicArray": Array [
                      ".a { background-color:gray /*TestComponent.sheet.root*/ }",
                    ],
                  },
                ],
              },
            },
          },
          Object {
            "classNameX": Object {
              "list": Array [
                Object {
                  "atomicArray": Array [
                    ".b { padding:10px /*classNameX*/ }",
                  ],
                },
              ],
            },
            "classes": null,
            "codeProps": Array [
              Object {
                "children": "Text",
              },
              null,
            ],
            "styleX": Object {
              "margin": 10,
            },
          },
          undefined,
          Object {},
        ],
        "props": Object {
          "children": "Text",
          "classNameX": Object {
            "padding": "10px",
          },
          "styleX": Object {
            "margin": 10,
          },
        },
        "propsCode": Object {
          "children": "Text",
          "classNameX": Object {
            "list": Array [
              Object {
                "atomicArray": Array [
                  ".b { padding:10px /*classNameX*/ }",
                ],
              },
            ],
          },
          "classes": Object {
            "root": Object {
              "list": Array [
                Object {
                  "atomicArray": Array [
                    ".a { background-color:gray /*TestComponent.sheet.root*/ }",
                  ],
                },
              ],
            },
          },
          "styleX": Object {
            "margin": 10,
          },
        },
        "uniqueId": "TestComponent *17",
      }
    }
  >
    <TestComponentCode
      classNameX={
        Object {
          "list": Array [
            Object {
              "atomicArray": Array [
                ".b { padding:10px /*classNameX*/ }",
              ],
            },
          ],
        }
      }
      classes={
        Object {
          "root": Object {
            "list": Array [
              Object {
                "atomicArray": Array [
                  ".a { background-color:gray /*TestComponent.sheet.root*/ }",
                ],
              },
            ],
          },
        }
      }
      styleX={
        Object {
          "margin": 10,
        }
      }
    >
      <ReactAny
        className="b a"
        style={
          Object {
            "margin": 10,
          }
        }
        trace={
          Array [
            ".b { padding:10px /*classNameX*/ }",
            ".a { background-color:gray /*TestComponent.sheet.root*/ }",
          ]
        }
      >
        traceLevel=5
      </ReactAny>
    </TestComponentCode>
  </InnerStateComponent>
</TestComponent>
`;
